### PR TITLE
Feat:  New components

### PR DESCRIPTION
**Pull request by bpolaszek**

### CODE REVIEW

File: src/ApplicationUI/Forms/Input/TextInput.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<template>
+  <input
+    :value="modelValue"
+    :class="classes"
+    @input="emit('update:modelValue', $event.target.value)"/>
```
Comment (neorel) [inline src/ApplicationUI/Forms/Input/TextInput.vue]: Missing `type="text"`

File: src/ApplicationUI/Forms/Input/TextInput.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+</template>
+
+<script setup>
+import { computed, ref } from 'vue';
+import { get } from '@vueuse/core';
+
+const emit = defineEmits(['update:modelValue']);
+const props = defineProps({
```
Comment (neorel) [inline src/ApplicationUI/Forms/Input/TextInput.vue]: Missing imports for `defineEmits` & `defineProps` ?

**Review by neorel — CHANGES_REQUESTED** [review]

File: src/ApplicationUI/Forms/Input/TextInput.vue
Comment (misaert) [inline src/ApplicationUI/Forms/Input/TextInput.vue:1]: As see together, this file can be deleted because of unused.

File: src/index.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 export { default as Alert } from './ApplicationUI/Alert/Alert.vue';
+export { default as Button } from './ApplicationUI/Forms/Buttons/Button.vue';
 export { default as ComboBox } from './ApplicationUI/Forms/ComboBoxes/ComboBox.vue';
+export { default as TextInput } from './ApplicationUI/Forms/Input/TextInput.vue';
```
Comment (misaert) [inline src/index.js]: To delete.

File: src/ApplicationUI/Forms/Buttons/Button.vue [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  noRing: {
+    type: Boolean,
+    default: false,
+  },
+  theme: {
+    type: String,
+    default: 'default',
+    validator: (value) => ['default', 'primary', 'danger', 'cancel'].includes(value),
+  },
+  icon: {
```
Comment (misaert) [inline src/ApplicationUI/Forms/Buttons/Button.vue]: Where is it used?

File: src/ApplicationUI/Forms/Buttons/Button.vue [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    type: String,
+    default: undefined,
+  },
+  iconRight: {
+    type: Boolean,
+    default: false,
+  },
+  iconType: {
```
Comment (misaert) [inline src/ApplicationUI/Forms/Buttons/Button.vue]: It seems unused.

File: src/ApplicationUI/Forms/Buttons/Button.vue [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    default: false,
+  },
+  noRing: {
+    type: Boolean,
+    default: false,
+  },
+  theme: {
+    type: String,
+    default: 'default',
+    validator: (value) => ['default', 'primary', 'danger', 'cancel'].includes(value),
```
Comment (misaert) [inline src/ApplicationUI/Forms/Buttons/Button.vue:53]: `reset` instead of `cancel` as name?

Comment (bpolaszek) [reply src/ApplicationUI/Forms/Buttons/Button.vue:53]: IMO `reset` is a sort of implementation of a more generic `cancel` - for instance a `cancel` button could be used to dismiss a modal or something like this, and would have the same design as a (form) `reset` button.

Comment (misaert) [reply src/ApplicationUI/Forms/Buttons/Button.vue:53]: I agree for this repository :+1:

File: src/ApplicationUI/Forms/Buttons/Button.vue (hunk 4 above)
Comment (bpolaszek) [reply src/ApplicationUI/Forms/Buttons/Button.vue]: Good catch, my original `Button` component has a built-in icon management, but relies on heroicons v2, so I removed its support here. Forgot to remove related props.

**Review by neorel — APPROVED** [review]

**Review by misaert — APPROVED** [review]

**Pull request merged**